app/models/manager_refresh/inventory_collection.rb

```diff
@@ -186,8 +186,12 @@ def lazy_find(manager_uuid, key: nil, default: nil)
       ::ManagerRefresh::InventoryObjectLazy.new(self, manager_uuid, :key => key, :default => default)
     end
 
+    def inventory_object_class
+      @inventory_object_class ||= Class.new(::ManagerRefresh::InventoryObject)
```

Comment:
Thats what I mean with the proper inheritance in my main comment. This is a somewhat hacky and magical way of defining new classes on the fly